### ISSUE TITLE
Getting 'fatal: Failed to enumerate all Git configuration entries [129]' when trying to clone a repo from Azure Dev Ops

### ISSUE THREAD

**Issue by simonneedham** [microsoft/Git-Credential-Manager-Core#320]
**Which version of GCM Core are you using?**
2.0.394-beta+3fc6791abf

**Which Git host provider are you trying to connect to?**

* [X] Azure DevOps
* [ ] Azure DevOps Server (TFS/on-prem)
* [ ] GitHub
* [ ] GitHub Enterprise
* [ ] Bitbucket
* [ ] Other - please describe

**Can you access the remote repository directly in the browser using the remote URL?**

From a terminal, run `git remote -v` to see your remote URL.

* [ ] Yes
* [ ] No, I get a permission error
* [X] No, for a different reason - please describe
I don't have a remote hooked up yet. Am trying to `git clone` a new local repo from Azure Dev Ops. I'm using the remote URL that Azure DevOps provides.
---

**_[Azure DevOps only]_ What format is your remote URL?**

* [ ] Not applicable
* [ ] https://dev.azure.com/`{org}`/...
* [X] https://`{org}`@dev.azure.com/`{org}`/...
* [ ] https://`{org}`.visualstudio.com/...

**_[Azure DevOps only]_ If the account picker shows more than one identity as you authenticate, check that you selected the same one that has access on the web.**

* [X] Not applicable
* [ ] I only see one identity
* [ ] I checked each identity and none worked

---

**Expected behavior**
GCM-Core pops up a window for me to enter my auth details.

**Actual behavior**
No auth window, error appears in console. Whatever I do I'm not able to figure out how to convince git to use the Git Credential Manager Core. I've tried manual configuration and running `git config --global credential.helper manager-core` to configure GCM.

```git clone https://{org name}@dev.azure.com/{org name}/{project name}/_git/{repo name}
Cloning into '{repo name}'...
fatal: Failed to enumerate all Git configuration entries [129]
error: unknown option `show-scope'
usage: git config [<options>]
```

My `.gitconfig` currently looks like this:
```
[user]
	name = {my name}
	email = {a randomised github email address}
[credential]
	helper = 
	helper = /usr/local/share/gcm-core/git-credential-manager-core
[credential "https://dev.azure.com"]
	useHttpPath = true
```
which is after running a `git credential-manager-core configure`

**Logs**

Set the environment variables `GCM_TRACE=1` and `GIT_TRACE=1` and re-run your Git command. Review and redact any private information and attach the log.

The log of the `git clone` attempt looks like this:

```
 {user name}@{mac name}  ~  git clone https://{org name}@dev.azure.com/{org name}/{project name}/_git/{repo name}
14:44:06.966165 git.c:419               trace: built-in: git clone https://{org name}@dev.azure.com/{org name}/{project name}/_git/{repo name}
Cloning into '{repo name}'...
14:44:06.977228 run-command.c:643       trace: run_command: git-remote-https origin https://{org name}@dev.azure.com/{org name}/{project name}/_git/{repo name}
14:44:07.208765 run-command.c:643       trace: run_command: 'git credential-osxkeychain get'
14:44:07.223395 git.c:670               trace: exec: git-credential-osxkeychain get
14:44:07.224189 run-command.c:643       trace: run_command: git-credential-osxkeychain get
14:44:07.241814 run-command.c:643       trace: run_command: 'git credential-manager-core get'
14:44:07.253224 git.c:670               trace: exec: git-credential-manager-core get
14:44:07.253789 run-command.c:643       trace: run_command: git-credential-manager-core get
14:44:07.384673 ...er/Application.cs:84 trace: [RunInternalAsync] Version: 2.0.394.50751
14:44:07.398377 ...er/Application.cs:85 trace: [RunInternalAsync] Runtime: .NET 5.0.4
14:44:07.398424 ...er/Application.cs:86 trace: [RunInternalAsync] Platform: macOS (x86-64)
14:44:07.398437 ...er/Application.cs:87 trace: [RunInternalAsync] AppPath: /usr/local/share/gcm-core/git-credential-manager-core
14:44:07.398515 ...er/Application.cs:88 trace: [RunInternalAsync] Arguments: get
14:44:07.463794 ...GitCommandBase.cs:35 trace: [ExecuteAsync] Start 'get' command...
14:44:07.476711 ...GitCommandBase.cs:49 trace: [ExecuteAsync] Detecting host provider for input:
14:44:07.477698 ...GitCommandBase.cs:50 trace: [ExecuteAsync] 	protocol=https
14:44:07.477724 ...GitCommandBase.cs:50 trace: [ExecuteAsync] 	host=dev.azure.com
14:44:07.477731 ...GitCommandBase.cs:50 trace: [ExecuteAsync] 	path={org name}/{project name}/_git/{repo name}
14:44:07.477736 ...GitCommandBase.cs:50 trace: [ExecuteAsync] 	username={org name}
14:44:07.485896 ...Configuration.cs:133 trace: [Enumerate] Failed to enumerate config entries (exit=129, level=All)
fatal: Failed to enumerate all Git configuration entries [129]
14:44:07.484351 git.c:419               trace: built-in: git config --null --list --show-scope
error: unknown option `show-scope'
usage: git config [<options>]


Password for 'https://{org name}@dev.azure.com/{org name}/{project name}/_git/{repo name}': 
14:44:10.242626 run-command.c:643       trace: run_command: 'git credential-osxkeychain erase'
14:44:10.247592 git.c:670               trace: exec: git-credential-osxkeychain erase
14:44:10.247997 run-command.c:643       trace: run_command: git-credential-osxkeychain erase
14:44:10.262107 run-command.c:643       trace: run_command: 'git credential-manager-core erase'
14:44:10.266678 git.c:670               trace: exec: git-credential-manager-core erase
14:44:10.267116 run-command.c:643       trace: run_command: git-credential-manager-core erase
14:44:10.353643 ...er/Application.cs:84 trace: [RunInternalAsync] Version: 2.0.394.50751
14:44:10.364088 ...er/Application.cs:85 trace: [RunInternalAsync] Runtime: .NET 5.0.4
14:44:10.364120 ...er/Application.cs:86 trace: [RunInternalAsync] Platform: macOS (x86-64)
14:44:10.364125 ...er/Application.cs:87 trace: [RunInternalAsync] AppPath: /usr/local/share/gcm-core/git-credential-manager-core
14:44:10.364153 ...er/Application.cs:88 trace: [RunInternalAsync] Arguments: erase
14:44:10.407605 ...GitCommandBase.cs:35 trace: [ExecuteAsync] Start 'erase' command...
14:44:10.414480 ...GitCommandBase.cs:49 trace: [ExecuteAsync] Detecting host provider for input:
14:44:10.415459 ...GitCommandBase.cs:50 trace: [ExecuteAsync] 	protocol=https
14:44:10.415477 ...GitCommandBase.cs:50 trace: [ExecuteAsync] 	host=dev.azure.com
14:44:10.415482 ...GitCommandBase.cs:50 trace: [ExecuteAsync] 	path={org name}/{project name}/_git/{repo name}
14:44:10.415485 ...GitCommandBase.cs:50 trace: [ExecuteAsync] 	username={org name}
14:44:10.415489 ...GitCommandBase.cs:50 trace: [ExecuteAsync] 	password=********
14:44:10.423095 ...Configuration.cs:133 trace: [Enumerate] Failed to enumerate config entries (exit=129, level=All)
fatal: Failed to enumerate all Git configuration entries [129]
14:44:10.421753 git.c:419               trace: built-in: git config --null --list --show-scope
error: unknown option `show-scope'
usage: git config [<options>]


fatal: Authentication failed for 'https://{org name}@dev.azure.com/{org name}/{project name}/_git/{repo name}/'
```

**Comment by cotyhamilton** [comment]
I had the same issue earlier on macOS. It was working fine, cloned a new repo this morning, then I ran a brew upgrade on gcm core and it produced your issue.

I was using git that shipped with xCode (sorry I no longer have version) and I got everything to start working after several reinstalls and git configs later by just installing a newer version of git.

Working with below configuration:

```
➜ git --version
git version 2.31.1

➜ git credential-manager-core --version
2.0.394-beta+3fc6791abf

➜ git config --list --show-origin
file:/usr/local/etc/gitconfig   credential.helper=osxkeychain
file:/Users/redacted/.gitconfig  user.name=Hamilton, Coty
file:/Users/redacted/.gitconfig  user.email=redacted
file:/Users/redacted/.gitconfig  credential.helper=
file:/Users/redacted/.gitconfig  credential.helper=/usr/local/share/gcm-core/git-credential-manager-core
file:/Users/redacted/.gitconfig  credential.https://dev.azure.com.usehttppath=true
```

**Comment by cotyhamilton** [comment]
Some info from a relevant issue regarding older versions of git: #313

**Comment by simonneedham** [comment]
@cotyhamilton You've nailed the issue, thank you 🙏. I didn't have brew installed until I needed to get the gcm installed on this mac. So I installed brew, it brought down the xcode command line tools (as well as lot of other stuff). I don't know what version of git I was using before but after all that I was running with 2.21.0. I've just used brew again to update git to 2.31.1 and everything is working as it should.

As a nice to have, might be worth considering adding a git version check to gcm to confirm a minimum version is installed. 🤷‍♂️
